pkg/controller/storagecluster/storagecluster.go

```diff
@@ -405,6 +406,54 @@ func (c *Controller) validateCloudStorageLabelKey(cluster *corev1.StorageCluster
 	return nil
 }
 
+func (c *Controller) runPreflightCheck(cluster *corev1.StorageCluster) error {
+	check, ok := cluster.Annotations[pxutil.AnnotationPreflightCheck]
+	if !ok || check == "skip" {
+		return nil
+	} else if check == "false" {
+		if cluster.Status.Phase == string(corev1.ClusterConditionTypePreflight)+string(corev1.ClusterOperationFailed) {
+			return fmt.Errorf("please make sure your cluster meet all prerequisites and rerun preflight check")
+		}
+		return nil
```

Comment:
shouldnt we return error here so it wont proceed with installation?

Reply:
the idea here is after each check, the annotation will be flipped to `false`, and we only return error when it says `PreflightFailed` in the status. otherwise we should proceed with the installation. Will have the preflight status in condition instead of Phase in future PRs.

Reply:
Create a ticket for the conditions, if not already created.